src/main/java/fi/aalto/cs/apluscourses/model/SubmissionResult.java

```diff
@@ -0,0 +1,70 @@
+package fi.aalto.cs.apluscourses.model;
+
+import static org.apache.commons.lang3.StringUtils.substringBetween;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import org.jetbrains.annotations.NotNull;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+public class SubmissionResult {
+
+  private int exerciseId;
+  private List<Integer> submissionIds;
+  private int submissionsCount;
+  //  private int maxSubmissionsCount;
+  private int pointsToPass;
+  private int maxPoints;
+  private int totalPoints;
+
+  /**
+   * A result of a single submission.
+   *
+   * @param exerciseId an id of the exercise.
+   * @param submissionIds an {@link List} of submission ids.
+   * @param submissionsCount an amount of submissions made.
+   * @param pointsToPass a required amount of points to pass the exercise.
+   * @param maxPoints a possible maximum amount of points for the exercise.
+   * @param totalPoints a total amount of points achieved for the exercise.
+   */
+  public SubmissionResult(int exerciseId, List<Integer> submissionIds, int submissionsCount,
```

Comment:
in general, there is a separate API endpoint to fetch the data for a single submission: https://plus.cs.aalto.fi/api/v2/submissions/2921874/ , but at this point decided not to implement a call for it